docs/profiling-django.md

```diff
@@ -0,0 +1,132 @@
+# Profiling Django 
+
+One tool we have when optimizing our code in Django is profiling. 
+Profiling helps us understand where our time is spent, and doing what. 
+It also helps us identify duplication of effort, 
+as well as targets for caching and optimization.
+
+Because our primary goal is 
+to ensure that page loads are as quick as possible for the public, 
+this document focuses on profile the Django request/response cycle: 
+view code, Wagtail page rendering, etc.
+
+This document covers the tools we use to 
+both generate and understand a profile.
+
+## Generating a profile from a URL
+
+To enable profile generation, 
+we use a lightweight middleware that wraps 
+[Python's builtin `cprofile` module](https://docs.python.org/3/library/profile.html), 
+[django-cprofile-middleware](https://github.com/omarish/django-cprofile-middleware). 
+This middleware is enabled by default for local development 
+when `DEBUG` is `True`.
+
+For any URL served by Django, 
+you can add the `?prof` querystring 
+(or `&prof` to an existing query string)
+in order to see a profile of the request/response cycle for that view. 
+
+For example, `http://localhost:8000/about-us/newsroom/?prof`, 
+might return something like:
+
+```
+         1579654 function calls (1575009 primitive calls) in 28.273 seconds
+
+   Ordered by: internal time
+   List reduced from 1108 to 500 due to restriction <500>
+
+   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
+    74952    2.349    0.000   17.292    0.000 utils.py:487(__getattr__)
+174882/174880    2.036    0.000    4.559    0.000 {built-in method builtins.isinstance}
+    58330    1.985    0.000    5.482    0.000 utils.py:328(__getattr__)
+    77732    1.900    0.000    5.027    0.000 utils.py:157(__getattr__)
+     2776    1.821    0.001   21.265    0.008 documents.py:89(init_prepare)
+     …
+```
+
+You can also sort this output using a `sort` field. 
+For example `http://localhost:8000/about-us/newsroom/?prof&sort=time`. 
```

Comment:
A more useful sort example might be something that isn't the default:

```suggestion
For example, to sort by call count instead of the default time sort: `http://localhost:8000/about-us/newsroom/?prof&sort=ncalls`. 
```